ddtrace/internal/datadog/profiling/stack_v2/include/vendored/utf8_validate.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+// From https://gist.github.com/ichramm/3ffeaf7ba4f24853e9ecaf176da84566
+inline static bool
+utf8_check_is_valid(const char* str, int len)
```

Comment:
Can we verify that this is needed for CPython >= 3.7? In theory, as far as stacks are concerned,  filenames, and names/qualnames should be instances of `str`

Reply:
Ah, yeah--underdocumented procedure, sorry.

I noticed very rarely during testing that sometimes non-unicode strings would be passed to libdatadog.  My hunch is that those are moments where one of two things happen
1. We're in the middle of reading a frame (well, code...) struct in echion and it gets over-written in a way that replaces the string with a valid pointer, but the target is a non-string.
2. We're in the middle of reading from a string location, but that string had been cleared and re-allocated to some other struct (maybe this is more likely).

This is rare, but currently in libdatadog it drops the whole sample.  Or it panics?  I forgot TBH.  What I'd like to do eventually is handle the difference in libdatadog, since I'm anxious about doing _multiple_ levels of encoding-validation in the first place.  For now, having a check like this burns some CPU, but ultimately allows us to retain more of the stack.

I'm fine either way (drop the sample efficiently or try to keep the sample inefficiently).  What do you think?

Reply:
Ah ok thanks for the extra context. I think it's fine to keep this for now then if we get the benefit of retaining more data.